corgi/api/views.py

```diff
@@ -380,12 +381,49 @@ class ComponentViewSet(ReadOnlyModelViewSet):  # TODO: TagViewMixin disabled unt
     filterset_class = ComponentFilter
     lookup_url_kwarg = "uuid"
 
+    def get_queryset(self):
+        mode = self.request.query_params.get("mode")
+        if mode == "latest":
+            # 'latest' mode only relevent in terms of a specific offering/product
+            # TODO - subject of a near-future PR will handle instances of when we have
+            #      multiple root components with the same component name yet different versions.
+            ofuri = self.request.query_params.get("ofuri")
+            if ofuri:
+                component_types = Q(type=Component.Type.SRPM) | Q(
+                    type=Component.Type.CONTAINER_IMAGE
+                )
+                results = (
+                    Component.objects.filter(
+                        component_types,
+                        product_streams__overlap=[ofuri],
```

Comment:
Should this not be `__contains`?

Reply:
I think overlap is better, as I found when investigating [CORGI-229](https://github.com/RedHatProductSecurity/component-registry/pull/173/files) that __contains was returning matches for arrayfield value substrings. For example a query for `o:redhat:rhel:8.6.0` was return matches for both `o:redhat:rhel:8.6.0` and `o:redhat:rhel:8.6.0.z`.

Reply:
That might be true for string matches, but this is an ArrayField lookup.